custom_components/tuya_local/devices/tuya_000004rime_weather_station.yaml

```diff
@@ -0,0 +1,674 @@
+            value: July
+            mask: 00FF0000000000
+          - dps_val: 8
+            value: August
+            mask: 00FF0000000000
+          - dps_val: 9
+            value: September
+            mask: 00FF0000000000
+          - dps_val: 10
+            value: October
+            mask: 00FF0000000000
+          - dps_val: 11
+            value: November
+            mask: 00FF0000000000
+          - dps_val: 12
+            value: December
+            mask: 00FF0000000000
+      - id: 101
+        optional: true
+        type: base64
+        name: Day
+        mapping:
+          - mask: "0000FF00000000"
+      - id: 101
+        optional: true
+        type: base64
+        name: Hour
+        mapping:
+          - mask: "000000FF000000"
+      - id: 101
+        optional: true
+        type: base64
+        name: Minute
+        mapping:
+          - mask: "00000000FF0000"
+      - id: 101
+        optional: true
+        type: base64
+        name: Second
+        mapping:
+          - mask: "0000000000FF00"
+      - id: 101
+        optional: true
+        type: base64
+        name: Weekday
+        mapping:
+          - mask: "000000000000FF"
+          - dps_val: 1
+            value: Monday
+            mask: 00FF0000000000
+          - dps_val: 2
+            value: Tuesday
+            mask: 00FF0000000000
+          - dps_val: 3
+            value: Wednesday
+            mask: 00FF0000000000
+          - dps_val: 4
+            value: Thursday
+            mask: 00FF0000000000
+          - dps_val: 5
+            value: Friday
+            mask: 00FF0000000000
+          - dps_val: 6
+            value: Saturday
+            mask: 00FF0000000000
+          - dps_val: 7
+            value: Sunday
+            mask: 00FF0000000000
+  - entity: switch
+    name: LCD night mode
+    icon: "mdi:weather-night"
+    category: config
+    dps:
+      - id: 129
+        type: boolean
+        name: switch
+      - id: 130
+        optional: true
+        type: base64
+        name: Payload
+      - id: 130
+        optional: true
+        type: base64
+        name: Start hour
+        mapping:
+          - mask: "FF000000"
+      - id: 130
+        optional: true
+        type: base64
+        name: Start minute
+        mapping:
+          - mask: "00FF0000"
+      - id: 130
+        optional: true
+        type: base64
+        name: End hour
+        mapping:
+          - mask: "0000FF00"
+      - id: 130
+        optional: true
+        type: base64
+        name: End minute
+        mapping:
+          - mask: "000000FF"
+## diagnostic section, base64 dps
+  - entity: sensor
+    category: diagnostic
+    name: Display settings
+    class: enum
+    dps:
+      - id: 108
+        optional: true
+        type: base64
+        name: sensor
+        mapping:
+          - mask: 0000000000000000000000
+          - dps_val: null
+            value: Unknown
+            mask: 0000000000000000000000
+          - value: OK
```

Comment:
not sure that this part is correct, the idea is show Unknow if value is null, otherwise - OK

Reply:
Mask should have some bits set.  If you want to check for all 00, then you need a mask of all FF.

Reply:
Since there are only two values, this could also be a binary_sensor, but I am not actually sure what info this is conveying.